### PR TITLE
BaSD-MS-003: create APPLICATIONS logic

### PR DESCRIPTION
**Pull request by leimeter-joaquin**
-the new files are applications.js and applications.json
-there are 3 functions, create, list and delete
-Here is the [Postman collection](https://go.postman.co/workspace/My-Workspace~11351f1a-2294-4e2a-897a-6f318ac7aaef/collection/18238768-09ba3ff6-f3a7-40dd-a8a9-745a659f13ef) to test the controller (you have to use desktop app to request on local host).

### CODE REVIEW

File: controllers/applications.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    if (err) {
+      return res.status(400).send(err)
+    } 
+    return res.status(201).send(newApplication)
+  })
+}
+
+const deleteApplication = (req, res) => {
+  const filteredApplications = Applications.filter(app => app.id !== req.query.id)
+  const removedApplication = Applications.filter(app => app.id === req.query.id)
```
Comment (estefrare) [inline controllers/applications.js]: you can make both filters into a single one. It is more performant.

```
let removedApplication
const filteredApplications = Applications.filter(app => {
  if(app.id === req.query.id) {
    removedApplication = app
    return false
  }
  return true
})
```

File: index.js [hunk 2]
```diff
@@ -1,9 +1,16 @@
...
 const app = express();
 const PORT = 4000;
 
 app.get('/', (req, res) => {
     res.send('La Bestia does the best readmes ever');
 })
 
-app.listen(PORT, () => console.log(`Server runing on port ${PORT}`))
\ No newline at end of file
+//PPLICATIONS
```
Comment (estefrare) [inline index.js]: remove this comment, it is redundant

File: data/applications.json [hunk 3]
```diff
@@ -0,0 +1 @@
+[{"id":"1636326730936","idPosition":"123","idPostulant":"1234","date":"asd"},{"id":"1636326740982","idPosition":"123","idPostulant":"1234","date":"asd"},{"id":"1636326748741","idPosition":"123","idPostulant":"1234","date":"asd"},{"id":"1636326832155","idPosition":"123","idPostulant":"1234","date":"asd"},{"id":"1636326833316","idPosition":"123","idPostulant":"1234","date":"asd"},{"id":"1636330584663","idPosition":"123","idPostulant":"1234","date":"asd"},{"id":"1636330590612","idPosition":"123","idPostulant":"1234","date":"asdasdasdasdas"},{"id":"1636330763769","idPosition":"123","idPostulant":"1234","date":"asdasdasdasdas"},{"id":"1636330764563","idPosition":"123","idPostulant":"1234","date":"asdasdasdasdas"}]
```
Comment (estefrare) [inline data/applications.json]: format this JSON to make more easy to read

**Review by estefrare — CHANGES_REQUESTED** [review]

File: controllers/applications.js [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    }
+    return true
+  })
+
+  if(removedApplication.length === 0) res.status(404).send('application not found')
+  fs.writeFile('./data/applications.json', JSON.stringify(filteredApplications), {}, err => {
+    if (err) {
+      return res.status(400).send(err)
+    }
+    return res.status(200).send(removedApplication)
```
Comment (estefrare) [inline controllers/applications.js]: use status code 204.

File: controllers/applications.js [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+  let removedApplication
+  const filteredApplications = Applications.filter(app => {
+    if(app.id === req.query.id) {
+      removedApplication = app
+      return false
+    }
+    return true
+  })
+
+  if(removedApplication.length === 0) res.status(404).send('application not found')
```
Comment (estefrare) [inline controllers/applications.js]:
```
if(!removedApplication.length) {
  return res.status(404).send('application not found')
}
```

**Review by estefrare — CHANGES_REQUESTED** [review]

**Review by estefrare — APPROVED** [review]

**Review by ravegliamartin — APPROVED** [review]

**Pull request merged**